app/controllers/application_controller.rb

```diff
@@ -157,7 +157,7 @@ def alert_and_redirect_moderated
       flash.now[:warning] = "This is a draft note. Once you're ready, click <a class='btn btn-success btn-xs' href='/notes/publish_draft/#{@node.id}'>Publish Draft</a> to make it public. You can share it with collaborators using this private link <a href='#{@node.draft_url(request.base_url)}'>#{@node.draft_url(request.base_url)}</a>"
     elsif @node.status == 3 && (params[:token].nil? || (params[:token].present? && @node.slug.split('token:').last != params[:token]))
       page_not_found
-    elsif @node.status != 1 && @node.status != 3 && !(logged_in_as(['admin', 'moderator']))
+    elsif @node.status != 1 && @node.status != 3 && current_user&.id != @node.author.id && !(logged_in_as(['admin', 'moderator']))
```

Comment:
Hi @varun-b-g what do you think of the suggestions given by CodeClimate?
```suggestion
    elsif @node.status != 1 && @node.status != 3 && current_user&.id != @node.author.id && !logged_in_as(%w[admin moderator])
```
![Screenshot 2020-10-20 at 16 56 27](https://user-images.githubusercontent.com/7622875/96596285-388d3d80-12f5-11eb-8827-52928c3d2f2a.png)